### DISCUSSION TITLE
Commissioner petition time out an retry logic

### DISCUSSION THREAD

**Discussion by rretanubun** [openthread/openthread#9294] [Q&A] [answered]
Ref: https://github.com/openthread/openthread/blob/main/src/core/meshcop/commissioner.hpp#L496-L498

I am trying to confirm my understanding on `commissioner petitioning`

when we issue `commissioner start` it will switch from `disabled -> petitioning` and provided that there is [no other commissioner?] in the network, it will transition to `active`.

I tried testing using these steps:
1. Setup 2x FTD (TBR) that are `routers` and `neighbors` to each other on the same thread network
2. On `FTD1` I issue `commissioner start` and it transitions to `active`
3. On `FTD2` I issue `commissioner start` and I was expecting it to petition for a `while` using what I was expecting as the retry logic and then eventually it will go from `petitioning -> disabled`

Instead both FTD1 and FTD2 can be `active` at the same time (checked with `commissioner state`)

So I am obviously misunderstanding how `kPetition* flags` are supposed to work. Any hints are greatly appreciated.

**Comment by jwhui** [comment, marked as answer]
The Leader determines the identity of the Commissioner using the Commissioner ID and only fails a petition request if the Commissioner ID is different. The Commissioner ID currently defaults to "OpenThread Commissioner". You can change the Commissioner ID using the `commissioner id` CLI command.

**Reply by rretanubun** [reply]
Thanks @jwhui ! I am able to get further and have another clarification question

update for others: If I use `commissioner id` to make the commissioner id unique on both `FTD` I am able to experience that while a commissioner is `active` on one of the `FTD`, issuing `commissioner start` on the other `FTD` leads to this sequence:

```
ot-ctl commissioner start
commissioner start
Commissioner: petitioning
Done
> Commissioner: disabled
```

The time gap between `petitioning -> disabled` is quicker than I expect though, I was expecting the `petitioning`  to go on for what the `kPetition*` flags control (i.e. ~10 seconds) before getting to `disabled`. In my empirical observation the device gets to `disabled` almost right way (< 2 seconds).

**Reply by jwhui** [reply]
The Leader will immediately respond with a Reject if there is already an active Commissioner and the Commissioner ID does not match.

**Comment by rretanubun** [comment]
Got it, thanks for all the guidance! This time I won't be lazy and `self-close` my QA 😅
